akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -215,6 +217,13 @@ private[http] object StreamUtils {
       i ⇒ f(i) :: Nil
     }
 
+  /**
+   * Lifts the streams attributes into an element and passes them to the function for each passed through element.
+   * Similar idea than [[FlowOps.statefulMapConcat]] but for a simple map.
+   */
+  def statefulAttrsMap[T, U](functionConstructor: Attributes ⇒ T ⇒ U): Flow[T, U, NotUsed] =
```

Comment:
Actually, the idea was to pass the attributes only once as they don't change while the stream is running.